### ISSUE TITLE
[FR] open to run scripts before and after running code

### ISSUE THREAD

**Issue by andrewcrook** [twibiral/obsidian-execute-code#22]
Just a thought people may be using languages in particular setups or environment using tools  such as asdf, nx, nvm, pyenv etc or may need to set configuration up in the background such as move files, importing, database, downloads, package manager etc. 

You could get around this by allowing the user to run setup and teardown scripts before and after code runs.
you could have default scripts in the settings page and fields in the forntmatter.

**Comment by pspiagicw** [comment]
Quite a needed feature!

**Comment by twibiral** [comment]
I was thinking about two different implementations:

- Code to execute before / after each block is set in the settings for each language
- Per note: Code blocks with `pre-` or `post-` can be defined in a note and are executed before / after each block of the same language. e.g.:

```pre-cpp
// some cpp to be executed before each cpp block.
```
```post-python
// some python to be executed after each python block
```

**Comment by twibiral** [comment]
Solved in [0.15.0](https://github.com/twibiral/obsidian-execute-code/releases/tag/0.15.0).
